api/xds_protocol.rst

```diff
@@ -155,45 +266,116 @@ the resources in the :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>`
+the server must include the complete state of the world, meaning that all resources of the relevant type that are needed by the
+client must be included, even if they did not change since the last response. This means that if the server has previously sent
+100 resources and only one of them has changed, it must resend all 100 of them, even the 99 that were not modified.
+
+Deleting Resources
+^^^^^^^^^^^^^^^^^^
+
+In the incremental proocol variants, the server signals the client that a resource should be deleted via the
+:ref:`removed_resources <envoy_api_field_DeltaDiscoveryResponse.removed_resources>` field of the response. This tells the client
+to remove the resource from its local cache.
+
+In the SotW protocol variants, the criteria for deleting resources is more complex. For :ref:`Listener <envoy_api_msg_Listener>`
+and :ref:`Cluster <envoy_api_msg_Cluster>` resource types, if a previously seen resource is not present in a new response, that
+indicates that the resource has been removed, and the client must delete it; a response containing no resources means to delete
+all resources of that type. However, for other resource types, the API provides no mechanism for the server to tell the client
+that resources have been deleted; instead, deletions are indicated implicitly by parent resources being changed to no longer
+refer to a child resource. For example, when the client receives an LDS update removing a :ref:`Listener <envoy_api_msg_Listener>`
+that was previously pointing to :ref:`RouteConfiguration <envoy_api_msg_RouteConfiguration>` A, if no other :ref:`Listener
+<envoy_api_msg_Listener>` is pointing to :ref:`RouteConfiguration <envoy_api_msg_RouteConfiguration>` A, then the client may
+delete A. For those resource types, an empty :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>` is effectively a no-op
+from the client's perspective.
+
+Knowing When a Requested Resource Does Not Exist
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For :ref:`Listener <envoy_api_msg_Listener>` and :ref:`Cluster <envoy_api_msg_Cluster>` resource types, because each response
+needs to include all resources requested by the client, if a client requests a resource that does not exist, it can immediately
+tell this from the response.
+
+However, for other resource types, because each resource can be sent in its own response, there is no way to know from the
+next response whether the newly requested resource exists, because the next response could be an unrelated update for another
+resource that had already been subscribed to previously. As a result, clients are expected to use a timeout (recommended
+duration is 15 seconds) after sending a request for a new resource, after which they will consider the requested resource to
+not exist if they have not received the resource. In Envoy, this is done for :ref:`RouteConfiguration
```

Comment:
There is a timeout for all xDS calls, not just route configuration and EDS.

Reply:
Okay.  I assume that this is just to guard against the server failing to send back any response in a timely manner, and I've added a note about that.